cloud/amazon/ec2_vpc_peering.py

```diff
@@ -0,0 +1,596 @@
+            accept_attempted = map(self.attempt_auto_accept, pending)
+
+            changed = results_list_has_changed(accept_attempted)
+
+        results = self.find_valid_matching_connections()
+        return (changed, results)
+
+    def ensure_absent(self):
+        """
+        Ensure no connections are present as per module criteria,
+        delete them if there are any.
+        """
+        changed = False
+        results = list()
+
+        valid_connections = self.find_valid_matching_connections()
+
+        if not valid_connections:
+            return (changed, results)
+
+        else:
+            for connection in valid_connections:
+                if self.always_delete or \
+                        not self.util.is_pending_accept(connection):
+                    self.util.delete_peering_connection(connection.id)
+                else:
+                    self.util.reject_peering_connection(connection.id)
+            changed = True
+
+        results = self.find_matching_connections()
+        return (changed, results)
+
+    def list_all(self):
+        """
+        Return the list of all VPC peering connections.
+        """
+        all_connections = self.util.get_all_peering_connections()
+        return (False, all_connections)
+
+
+def connection_as_dict(connection):
+    """
+    Return a peering connection as dictionary.
+    """
+    if not connection:
+        return dict()
+
+    return {
+        'id': connection.id,
+        'expiration': connection.expiration_time or '',
+        'source_vpc_id':  connection.requester_vpc_info.vpc_id,
+        'source_vpc_cidr': connection.requester_vpc_info.cidr_block,
+        'peer_vpc_id': connection.accepter_vpc_info.vpc_id,
+        'peer_vpc_cidr': connection.accepter_vpc_info.cidr_block,
+        'status': connection.status_code,
+        'status_message': connection.status_message,
+        'tags': connection.tags
+    }
+
+
+def results_list_has_changed(results):
+    """
+    Given a list of results where each result is a tuple of the form
+    (changed, ...), [where changed is a boolean], return True if this group
+    has changed.
+    """
+    get_changed = itemgetter(0)
+    return any(filter(get_changed, results))
+
+
+def wait_until_condition(condition_check, interval=1):
+    """
+    Given a condition check as callable, sleep until the condition check
+    returns True
+    """
+    while True:
+        time.sleep(interval)
+        if condition_check():
+            break
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+            region = dict(required=True),
+            name = dict(required=False),
+            source_vpc = dict(required=False),
+            peer_vpc = dict(required=False, aliases=['dest_vpc']),
+            peer_owner_id = dict(required=False),
+            state = dict(
+                choices=['absent', 'present', 'active', 'pending', 'list'],
```

Comment:
I am willing to make the change. Do you have any example for fact module? I not familiar with ansible module development. Tried to find some examples in source code and devel maillist, but I didn't find anything. Thanks.